src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -736,6 +734,9 @@ void RegisterAppInterfaceRequest::SendRegisterAppInterfaceResponseToMobile(
     resumer.StartResumptionOnlyHMILevel(application);
   }
 
+  SendOnAppRegisteredNotificationToHMI(
+      *(application.get()), resumption, need_restore_vr);
```

Comment:
This results in the resumption commands (AddCommand, etc.) being sent before `OnAppRegistered` is sent to the HMI. This means that the HMI would need to add commands for an app that it doesn't know exists yet, which causes errors with the sample HMIs.